src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1472,15 +1480,20 @@ FixedwingPositionControl::control_auto_takeoff(const hrt_abstime &now, const flo
 		 * Update navigation: _runway_takeoff returns the start WP according to mode and phase.
 		 * If we use the navigator heading or not is decided later.
 		 */
+		Vector2f curr_pos_local{_local_pos.x, _local_pos.y};
+		Vector2f curr_wp_local = _global_local_proj_ref.project(curr_wp(0), curr_wp(1));
```

Comment:
this is done twice, once in each if condition -- could we move it out of the if statement and replace this line?
https://github.com/PX4/PX4-Autopilot/blob/a2164e20b1a0b58d29509eab519367099ca2948d/src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp#L1430-L1431

could also possibly do the prev wp there